src/bot.ts

```diff
@@ -106,14 +100,60 @@ Also, here are some links you might find useful:
   )
 }
 
-function _handleFetchDataError (error: Error): string {
-  log.error(error)
-  return 'N/A'
-}
-
-dfusionRepo.watchOrderPlacement({
-  onNewOrder (event) {
-    bot.sendMessage(channelId, formatNewOrders(event))
+dfusionService.watchOrderPlacement({
+  onNewOrder (order) {
+    const {
+      // owner,
+      buyToken,
+      sellToken,
+      validFrom,
+      validUntil,
+      // validFromBatchId,
+      // validUntilBatchId,
+      priceNumerator,
+      priceDenominator
+      // event
+    } = order
+
+    // Calculate the price
+    let price
+    if (buyToken.decimals >= sellToken.decimals) {
+      const precisionFactor = 10 ** (buyToken.decimals - sellToken.decimals)
+      price = priceNumerator.dividedBy(priceDenominator.multipliedBy(precisionFactor))
+    } else {
+      const precisionFactor = 10 ** (sellToken.decimals - buyToken.decimals)
+      price = priceNumerator.multipliedBy(precisionFactor).dividedBy(priceDenominator)
+    }
+
+    // Label for token
+    const sellTokenLabel = sellToken.symbol || sellToken.name || sellToken.address
+    const buyTokenLabel = buyToken.symbol || buyToken.name || buyToken.address
+
+    // Only display the valid from if the period hasn't started
+    const now = new Date()
+    let datesDescription = ''
+    if (validFrom > now) {
+      // The order is not active yet
+      datesDescription = `  - *Tradable*: \`${moment(validFrom).calendar()} GMT\`, \`${moment(validFrom).fromNow()}\`\n`
```

Comment:
I suggest a small refactor with a function such as `function buildDateString(label: string, d: date): string {}`

Reply:
Handled in different PR cause I need to leave and I want to merge this tomorrow :) 
https://github.com/gnosis/dex-telegram/issues/30